h/services/group.py

```diff
@@ -66,7 +60,29 @@ def create_open_group(self, name, userid, origins, description=None):
                             userid=userid,
                             description=description,
                             type_flags=OPEN_GROUP_TYPE_FLAGS,
-                            scopes=[GroupScope(origin=o) for o in origins],
+                            origins=origins,
+                            add_creator_as_member=False,
+                            )
+
+    def create_restricted_group(self, name, userid, origins, description=None):
+        """
+        Create a new restricted group.
+
+        A restricted group is one that anyone in the same authority can read but
+        only members can write.
+
+        :param name: the human-readable name of the group
+        :param userid: the userid of the group creator
+        :param description: the description of the group
+
+        :returns: the created group
+        """
+        return self._create(name=name,
+                            userid=userid,
+                            description=description,
+                            type_flags=RESTRICTED_GROUP_TYPE_FLAGS,
+                            origins=origins,
+                            add_creator_as_member=True,
```

Comment:
Oh, good! I'm glad this is here (auto-joining creator as member), at least for now. Sweet.